azext_iot/central/models/template.py

```diff
@@ -0,0 +1,57 @@
+# coding=utf-8
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from knack.util import CLIError
+
+
+class Template:
+    def __init__(self, template: dict):
+        self.template = template
+        try:
+            self.id = template.get("id")
+            self.name = template.get("displayName")
+            self.interfaces = self._extract_interfaces(template)
+            self.schema_names = self._extract_schema_names(self.interfaces)
+        except:
+            raise CLIError("Could not parse iot central device template.")
+
+    def get_schema(self, telemetry_name, interface_name=""):
+        # interface_name has been specified, do a pointed lookup
+        if interface_name:
+            interface = self.interfaces.get(interface_name, {})
+            return interface.get(telemetry_name)
+
+        # find first matching telemetry_name in any interface
+        for interface in self.interfaces.values():
+            schema = interface.get(telemetry_name)
+            if schema:
+                return schema
+
+        # not found
+        return None
+
+    def _extract_interfaces(self, template: dict) -> dict:
+        try:
+            dcm = template.get("capabilityModel", {})
```

Comment:
This would support dtdl version 1 preview, but not version 2 - but that is currently a service constraint yes?

Reply:
Yes correct, this code will be updated if/when we upgrade the dtdl support we have in central